api/reviews_api.py

```diff
@@ -57,7 +57,7 @@ def do_get(self, **kwargs) -> dict[str, list[dict[str, Any]]]:
     # Note: We assume that anyone may view approvals.
     votes = Vote.get_votes(feature_id=feature_id, gate_id=gate_id)
     dicts = [converters.vote_value_to_json_dict(v) for v in votes]
-    return GetVotesResponse.from_dict({'votes': dicts}).to_dict()
+    return GetVotesResponse.model_construct(votes=dicts).to_dict()
```

Comment:
```suggestion
    return GetVotesResponse.from_dict({'votes': dicts}).to_dict()
```

model_construct takes in the expected object type for each argument. (votes should expect a variable of type `List[Votes]`). However, this is a special case the code does this special conversion to dictionaries first. So instead, you can use the class method from_dict.

Reply:
This did not work either. This indeed helps convert the dict to model and then using to_dict() to convert it back, but it seems like the serialization & deserialization during the process changed the order of stuff we put in the dict and therefore differ from the test case. How would you suggest circumventing this situation?

Reply:
What error are you seeing? When I tried this suggestion earlier, the test started passing.

Reply:
i got  
`FAIL: test_do_get__success (api.reviews_api_test.GatesAPITest.test_do_get__success)
Handler retrieves all gates associated with a given feature.
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/mock.py", line 1375, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mark/chromium-dashboard/api/reviews_api_test.py", line 349, in test_do_get__success
    self.assertEqual(actual, expected)
AssertionError: {'gat[126 chars]e', 'state': 1, 'assignee_emails': [], 'additi[125 chars]ne}]} != {'gat[126 chars]e', 'escalation_email': None, 'state': 1, 'req[265 chars]']}]}
Diff is 674 characters long. Set self.maxDiff to None to see it.` Looks like the order is still messed up. The only change I made was to undo the changes in the current commit to `return GetVotesResponse.from_dict({'votes': dicts}).to_dict()`

Reply:
That failing test is for a different method than this one.  This test should be passing now. Could you verify that?

The GatesAPITest.test_do_get__success test is for another class in the same file. I verified that  GatesAPITest.test_do_get__success is failing for me too.

```py
  @mock.patch('internals.approval_defs.get_approvers')
  def test_do_get__success(self, mock_get_approvers):
    """Handler retrieves all gates associated with a given feature."""
    mock_get_approvers.return_value = ['reviewer1@example.com']

    self.maxDiff = None
    with test_app.test_request_context(self.request_path):
      actual = self.handler.do_get(feature_id=self.feature_id)

    expected = {
        "gates": [
            {
                "id": 1,
                "feature_id": self.feature_id,
                "stage_id": 1,
                "gate_type": 1,
                "team_name": "API Owners",
                "gate_name": "Intent to Prototype",
                "escalation_email": None,
                "state": 1,
                # "requested_on": None,
                # "responded_on": None,
                "assignee_emails": [],
                # "next_action": None,
                "additional_review": False,
                'slo_initial_response': 5,
                # 'slo_initial_response_took': None,
                # 'slo_initial_response_remaining': None,
                'possible_assignee_emails': ['reviewer1@example.com'],
            },
        ],
        }

    self.assertEqual(actual, expected)
```

You will see that it is not an order issue but it is removing the null values instead of defaulting to None. That would be the first thing I would look into.

Reply:
Seems like if you need to add `"nullable: true"` to those fields I commented out in my above comment (similar to [escalation_email](https://github.com/GoogleChrome/chromium-dashboard/blob/22b6a18127c163d6ae3a5f338543dc7dc812d21b/openapi/api.yaml#L1045-L1047)). Then, it will work. Then you can revert the commented lines I made in the test.

Reply:
pasted the wrong test.. my bad. Adding `nullable: true` gets the test passed. I'll modify other specs as well to see if all tests could pass. Thanks!